### PR TITLE
Elasticsearch: Add User-Agent

### PR DESCRIPTION
**Pull request by vidok**
## What

This PR adds a proper User-Agent header to Elasticsearch source and destination connectors. Also, it bumps the Elasticsearch client version. 

Note: Upgrading Elasticsearch client to 8.x requires a lot of work so I will probably address this in a separate PR. 

## How

Set User-Agent to Elasticsearch client. 

## 🚨 User Impact 🚨
No user impact

### CODE REVIEW

File: airbyte-integrations/connectors/destination-elasticsearch/src/main/java/io/airbyte/integrations/destination/elasticsearch/ElasticsearchConnection.java
Comment (natikgadzhi) [inline airbyte-integrations/connectors/destination-elasticsearch/src/main/java/io/airbyte/integrations/destination/elasticsearch/ElasticsearchConnection.java:123]: Looks like that would return the name of the connector alongside the version, i.e. `airbyte-destination-elasticsearch@0.1.7`, right? Should the function be called something like `makeConnectorUAString`, if it's also only used for the user agent?

File: airbyte-integrations/connectors/source-elasticsearch/src/main/java/io/airbyte/integrations/source/elasticsearch/ElasticsearchConnection.java
Comment (natikgadzhi) [inline airbyte-integrations/connectors/source-elasticsearch/src/main/java/io/airbyte/integrations/source/elasticsearch/ElasticsearchConnection.java:100]: Same as the destination one — naming perhaps can be more explicit, `makeConnectorUAString`? Not attached to a specific name.

**Review by natikgadzhi — COMMENTED** [review]
Thank you for contributing, @vidok! The change looks good to me, but I'll leave it to @marcosmarxm to review and run this by destinations team.

File: airbyte-integrations/connectors/destination-elasticsearch/src/main/java/io/airbyte/integrations/destination/elasticsearch/ElasticsearchConnection.java
Comment (vidok) [reply airbyte-integrations/connectors/destination-elasticsearch/src/main/java/io/airbyte/integrations/destination/elasticsearch/ElasticsearchConnection.java:123]: Yeah. This makes sense to me. I will rename the function. 

Thanks 👍🏻

**Comment by vidok** [comment]
> Thank you for contributing, @vidok! The change looks good to me, but I'll leave it to @marcosmarxm to review and run this by destinations team.

@marcosmarxm I wonder if you have some time to have a look at this PR. Thanks in advance.

**Comment by seanstory** [comment]
Bump @natikgadzhi @marcosmarxm can we get this in?

Our team at Elastic is very interested to have metics to understand how popular this sink is. With the data to back up our assumptions, we may be able to justify investing cycles to help contribute to and maintain this. But a first step is proving out how widely this sink is used by our customers, and that means being able to get this user-agent change merged.

**Comment by serenachou** [comment]
@natikgadzhi  @marcosmarxm  hi there, any chance we could review this?

**Comment by marcosmarxm** [comment]
Folks sorry the lack of updates and keep this open for a long time. The current tests for ElasticSearch is broken and I need to fix them. After that I'll make sure to merge the contribution. I hope to have this done in next two weeks. If you need this change you can pull the branch and build the connector locally asap.

**Comment by serenachou** [comment]
@marcosmarxm  thank you so much, we really appreciate the update!

File: airbyte-integrations/connectors/source-elasticsearch/build.gradle
Comment (evantahler) [inline airbyte-integrations/connectors/source-elasticsearch/build.gradle:23]: Merge conflict

**Review by evantahler — CHANGES_REQUESTED** [review]

File: airbyte-integrations/connectors/source-elasticsearch/build.gradle
Comment (vidok) [reply airbyte-integrations/connectors/source-elasticsearch/build.gradle]: Thanks! Fixed.

**Comment by evantahler** [comment]
@marcosmarxm - as this is a community connector, if this passes the tests (and the tests are fixed - thank you!), 👍 from me!

**Comment by serenachou** [comment]
Hey there @marcosmarxm  is there anything we can do to assist the merge now that it's gotten that nice thumbs up from Evan?

File: airbyte-integrations/connectors/destination-elasticsearch/metadata.yaml
Comment (evantahler) [inline airbyte-integrations/connectors/destination-elasticsearch/metadata.yaml:5]: There should be a similar file to update in the destination-elasticseach-strict-encrypt directory as well

**Comment by marcosmarxm** [comment]
Most integration tests are failing @serenachou you can try to fix them or wait for the team. But at the moment it isn't in the roadmap fix them :(

**Comment by vidok** [comment]
> Most integration tests are failing @serenachou you can try to fix them or wait for the team. But at the moment it isn't in the roadmap fix them :(

Are those test can run by `gradlew :airbyte-integrations:connectors:destination-elasticsearch` command? I found it [here](https://docs.airbyte.com/connector-development/testing-connectors/).

**Comment by marcosmarxm** [comment]
> > Most integration tests are failing @serenachou you can try to fix them or wait for the team. But at the moment it isn't in the roadmap fix them :(
> 
> Are those test can be run by `gradlew :airbyte-integrations:connectors:destination-elasticsearch` command? I found it [here](https://docs.airbyte.com/connector-development/testing-connectors/).
> 
> Currently, this command is failing with:
> 
> ```
> Cannot locate tasks that match ':airbyte-integrations:connectors:destination-elasticsearch' as task 'destination-elasticsearch' not found in project ':airbyte-integrations:connectors'.
> ```

You can use the `airbyte-ci` today @vidok

**Pull request closed without merging**
